test/e2e/curated_packages_test.go

```diff
@@ -4,32 +4,177 @@
+) error {
+	end := time.Now().Add(timeout)
+	if !strings.HasPrefix(jsonpath, "jsonpath") {
+		jsonpath = fmt.Sprintf("jsonpath='%s'", jsonpath)
+	}
+	command := []string{}
+	for time.Now().Before(end) {
+		out, err := test.KubectlClient.Execute(ctx, "get", "-n", namespace,
+			"--kubeconfig="+kubeconfig.FromClusterName(test.ClusterName),
+			"-o", jsonpath, resource)
+		outStr := out.String()
+		trimmed := strings.Trim(outStr, "'")
+		allPredicates := true
+		for _, f := range predicates {
+			allPredicates = allPredicates && f(trimmed, err)
+		}
+		if allPredicates {
+			return nil
+		}
+		time.Sleep(2 * time.Second)
+	}
+	return fmt.Errorf(
+		"timed out waiting for resource: %s [namespace: %s, jsonpath: %s, timeout: %s]",
+		command,
+		namespace,
+		jsonpath,
+		timeout,
+	)
+}
+
+func WaitForDaemonset(
+	test *framework.ClusterE2ETest,
+	ctx context.Context,
+	daemonsetName string,
+	namespace string,
+	numberOfNodes int,
+	timeout time.Duration,
+) error {
+	return WaitForResource(
+		test,
+		ctx,
+		"daemonset/"+daemonsetName,
+		namespace,
+		"{.status.numberAvailable}",
+		timeout,
+		NoErrorPredicate,
+		IntEqualPredicate(numberOfNodes),
+	)
+}
+
+// Hackish way to get the latest bundle. This assumes no bundle is created outside of the normal PBC bundle fetch timer.
+// This should be modified to get the bundle from the previous build step and use that only.
+func WaitForLatestBundleToBeAvailable(
+	test *framework.ClusterE2ETest,
+	ctx context.Context,
+	timeout time.Duration,
+) error {
+	return WaitForResource(test, ctx, "packagebundle", "eksa-packages", "{.items[0]}", timeout, NoErrorPredicate)
+}
+
+func WaitForPackageToBeInstalled(
+	test *framework.ClusterE2ETest,
+	ctx context.Context,
+	packageName string,
+	timeout time.Duration,
+) error {
+	//--for=jsonpath isn't supported in v1.22. Update once it's supported
+	//_, err = test.KubectlClient.Execute(
+	//    ctx, "wait", "--timeout", "1m",
+	//    "--for", "jsonpath='{.status.state}'=installed",
+	//    "package", packagePrefix, "--kubeconfig", kubeconfig,
+	//    "-n", "eksa-packages",
+	//)
+	return WaitForResource(
+		test,
+		ctx,
+		"package/"+packageName,
+		EksaPackagesNamespace,
+		"{.status.state}",
+		timeout,
+		NoErrorPredicate,
+		StringMatchPredicate("installed"),
+	)
+}
+
+func UpgradePackages(test *framework.ClusterE2ETest, bundleVersion string) {
+    test.RunEKSA([]string{"upgrade", "packages", "--bundle-version="+bundleVersion})
+}
+
+func GetLatestBundleFromCluster(test *framework.ClusterE2ETest) (string, error) {
+	bundleBytes, err := test.KubectlClient.ExecuteCommand(
+		context.Background(),
+		"get",
+		"packagebundle",
+		"-n", "eksa-packages",
+		"--kubeconfig="+kubeconfig.FromClusterName(test.ClusterName),
+		"-o", "jsonpath='{.items[0].metadata.name}'",
+	)
+	if err != nil {
+		return "", err
+	}
+	bundle := bundleBytes.String()
+	return strings.Trim(bundle, "'"), nil
+}
+
+func TestPackagesInstallSimpleFlow(t *testing.T) {
```

Comment:
What is this testing? It doesn't fit with the established naming pattern of the other hello-eks-anywhere tests. If it's testing something that isn't already covered, even just a comment explaining what it's testing would be good.

Either way, if this is a test we want to keep, could this please be renamed "TestCPackagesInstallSimpleFlow"? That way it can get picked up by the "all" target in [the Makefile](https://github.com/aws/eks-anywhere/blob/16f4ebe91cc683f70d0470957cb900e93c88e6f5/Makefile#L427).

Reply:
Seems like it was resurrected from a rebase.